lib/reporter.js

```diff
@@ -51,14 +51,41 @@ class Reporter {
         this.collectionRunOptions = collectionRunOptions;
         this.options = options;
         this.collectionPath = utils.getCollectionPath(this.collectionRunOptions.workingDir);
+        this.launchLogs = [];
+        this.suiteLogs = [];
+        this.logs = [];
 
+        emitter.on('console', this.onConsole.bind(this));
         emitter.on('start', this.onStart.bind(this));
         emitter.on('beforeRequest', this.onBeforeRequest.bind(this));
         emitter.on('beforeTest', this.onBeforeTest.bind(this));
         emitter.on('beforeDone', this.onBeforeDone.bind(this));
         emitter.on('done', this.onDone.bind(this));
     }
 
+    onConsole (err, args) {
+        if (err) {
+            throw err;
+        }
+
+        const type = args.messages && args.messages[0];
+
+        switch (type) {
+            case 'launch':
+                if (this.launchLogs !== null) {
+                    args.messages.slice(1).forEach(message => this.launchLogs.push({ level: args.level, message }));
```

Comment:
Is it possible to collect the time of each log message and send it later to the API? Could we restore the sequence of log messages? It seems a user will expect to have logs in proper order in the results.

Reply:
Thank you for the comment. I added the additional parameter `time`.